pkg/boot/kexec/genpurg.go

```diff
@@ -0,0 +1,95 @@
+//go:build ignore
+
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+func main() {
+	var fail bool
+	f, err := os.Create("asm.go")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			log.Fatal("Closing %v: %v", f.Name(), err)
+		}
+	}()
+
+	if _, err := fmt.Fprintf(f, "package kexec\nvar purgatories = map[string]*purgatory {\n"); err != nil {
+		log.Fatal(err)
+	}
+	d, err := ioutil.TempDir("", "kexecgen")
+	if err != nil {
+		log.Fatalf("%v", err)
+	}
+
+	for _, asm := range asms {
+		// assemble, then create the hexdump of the elf.
+		//defer os.RemoveAll(d)
```

Comment:
Does this need to stay here?

Reply:
will fix.

Reply:
Done: https://github.com/u-root/u-root/pull/2317